### PR TITLE
Handle expired download URL with support for retries

### PR DESCRIPTION
**Pull request by manjanac**
Hey there! So you want to contribute to a Firebase SDK?
Before you file this pull request, please read these guidelines:

### Discussion

  * Read the contribution guidelines (CONTRIBUTING.md).
  * If this has been discussed in an issue, make sure to link to the issue here.
    If not, go file an issue about this **before creating a pull request** to discuss.

### Testing

  * Make sure all existing tests in the repository pass after your change.
  * If you fixed a bug or added a feature, add a new test to cover your code.

### API Changes

  * At this time we cannot accept changes that affect the public API.  If you'd like to help
    us make Firebase APIs better, please propose your change in a feature request so that we
    can discuss it together.

### CODE REVIEW

File: FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift
Comment (ryanwilson) [inline FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift:130]: For readability in the else statement and less indenting (I had to carefully measure where this `else` ended), I wonder if a `guard` statement would be better here too? Or can you not have a nested guard? Haven't tried that before.

```
guard remoteModelInfo.urlExpiryTime > currentDateTime else {
  downloadStatus = .failed
  downloadHandlers.completion(.......)
  return
}

// URL has not expired, try the download again.
```

File: FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift
Comment (ryanwilson) [inline FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift:132]: Same here for guard, if that's allowed in order to reduce the nested statements.

Comment (manjanac) [reply FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift]: Yes it is! I made the change.

File: FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift
Comment (manjanac) [reply FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift]: Good call, changed.

**Review by ryanwilson — APPROVED** [review]
Nice! Thanks for making the change. Please wait for CI, but LGTM

**Pull request merged**
